cpp/log/cert_checker.cc

```diff
@@ -292,26 +293,27 @@ Status CertChecker::GetTrustedCa(CertChain* chain) const {
        it != issuer_range.second; ++it) {
     const Cert* issuer_cand = it->second;
 
-    Cert::Status ok = subject->IsSignedBy(*issuer_cand);
-    if (ok == Cert::UNSUPPORTED_ALGORITHM) {
+    StatusOr<bool> signed_by_issuer = subject->IsSignedBy(*issuer_cand);
+    if (signed_by_issuer.status().CanonicalCode() == Code::UNIMPLEMENTED) {
```

Comment:
It's not an easy one to answer. In some cases the low level error can be
translated to something more meaningful at a higher level. OTOH you can end
up losing the actual cause of the problem by returning something generic.

On 4 September 2015 at 11:37, Pierre Phaneuf <notifications@github.com>
wrote:

> In cpp/log/cert_checker.cc
> <https://github.com/google/certificate-transparency/pull/905#discussion_r38739808>
> :
>
> > @@ -292,26 +293,27 @@ Status CertChecker::GetTrustedCa(CertChain* chain) const {
> >         it != issuer_range.second; ++it) {
> >      const Cert* issuer_cand = it->second;
> >
> > -    Cert::Status ok = subject->IsSignedBy(*issuer_cand);
> > -    if (ok == Cert::UNSUPPORTED_ALGORITHM) {
> > +    StatusOr<bool> signed_by_issuer = subject->IsSignedBy(*issuer_cand);
> > +    if (signed_by_issuer.status().CanonicalCode() == Code::UNIMPLEMENTED) {
>
> I wonder if we shouldn't just pass back the underlying status, or if in
> fact, this sort of translation is good (to control better what the end user
> see in the HTTP responses)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/certificate-transparency/pull/905/files#r38739808>
> .
>